src/rule_definition_tools.jl

```diff
@@ -300,8 +300,25 @@ macro non_differentiable(sig_expr)
 
     primal_name, orig_args = Iterators.peel(sig_expr.args)
 
+    local primal_name_sig
```

Comment:
Could we put this bit in a separate function?

Reply:
Yes, I can do that